app/src/main/java/com/infomaniak/drive/ui/fileShared/FileSharedListFragment.kt

```diff
@@ -122,8 +162,50 @@ class FileSharedListFragment : FileListFragment() {
         )
     }
 
+    private fun downloadAllFiles() {
+        fileSharedViewModel.rootSharedFile.value?.let { file ->
+            drivePermissions?.let { permissions -> requireContext().downloadFile(permissions, file) }
+        }
+    }
+
+    private fun onDriveAndFolderSelected(data: Intent?) {
+        val destinationDriveId = data?.getIntExtra(DESTINATION_DRIVE_ID_KEY, DEFAULT_ID) ?: DEFAULT_ID
+        val destinationFolderId = data?.getIntExtra(DESTINATION_FOLDER_ID_KEY, DEFAULT_ID) ?: DEFAULT_ID
+
+        if (data == null || destinationDriveId == DEFAULT_ID || destinationFolderId == DEFAULT_ID) {
+            showSnackbar(RCore.string.anErrorHasOccurred)
+        } else {
+            fileSharedViewModel.importFilesToDrive(
+                destinationDriveId = destinationDriveId,
+                destinationFolderId = destinationFolderId,
+                fileIds = multiSelectManager.selectedItemsIds.toList(),
+                exceptedFileIds = multiSelectManager.exceptedItemsIds,
+            )
+        }
+    }
+
+    private fun setMainButton(importButton: MaterialButton) {
+        importButton.setOnClickListener {
+            if (AccountUtils.currentDriveId == -1) {
```

Comment:
Does this really happen?

Reply:
Yes if you open the link without being connected, you don't have a drive